File: datamodel/src/main/java/datamodel/clinical/PriorSecondPrimary.java
package datamodel.clinical;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PriorSecondPrimary {

    @NotNull
    public abstract String tumorLocation();

    @NotNull
    public abstract String tumorSubLocation();

    @NotNull
    public abstract String tumorType();

    @NotNull
    public abstract String tumorSubType();

    @NotNull
    public abstract Set<String> doids();

    @Nullable
    public abstract Integer diagnosedYear();

    @Nullable
    public abstract Integer diagnosedMonth();

    @NotNull
    public abstract String treatmentHistory();

    @Nullable
    public abstract Integer lastTreatmentYear();

    @Nullable
    public abstract Integer lastTreatmentMonth();

    public abstract boolean isActive();

}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/PriorSecondPrimary.kt
package com.hartwig.actin.clinical.datamodel

data class PriorSecondPrimary(
    val tumorLocation: String,
    val tumorSubLocation: String,
    val tumorType: String,
    val tumorSubType: String,
    val doids: Set<String> = emptySet(),
    val diagnosedYear: Int? = null,
    val diagnosedMonth: Int? = null,
    val treatmentHistory: String,
    val lastTreatmentYear: Int? = null,
    val lastTreatmentMonth: Int? = null,
    val status: TumorStatus
)
